### PR TITLE
Remove Open Frameworks 

### PR DESCRIPTION
**Pull request by dterzic**
IOSP-40 Modify ofxGeneric project to build without openFrameworks.

### CODE REVIEW

File: src/iOS/ofxGenericAppDelegate.h [hunk 1]
```diff
@@ -8,10 +8,11 @@
 
 #pragma once
 
-#include "ofMain.h"
+// IOSP-40 // #include "ofMain.h"
+
 class ofxAppGenericWindow;
 
-#if TARGET_OS_IPHONE
+// IOSP-40 // #if TARGET_OS_IPHONE
```
Comment (erynofwales) [inline src/iOS/ofxGenericAppDelegate.h]: Can we just remove this `#if` and the `#endif` below?

File: src/iOS/ofxGenericAppDelegate.mm [hunk 2]
```diff
@@ -8,20 +8,13 @@
 
 #import "ofxGenericAppDelegate.h"
 #import "ofxGenericApp.h"
-#import "ofMain.h"
+// IOSP-40 // #import "ofMain.h"
```
Comment (erynofwales) [inline src/iOS/ofxGenericAppDelegate.mm]: Same here. Can this line just be removed?

File: src/ofxGenericApp.cpp [hunk 3]
```diff
@@ -87,7 +90,7 @@ ofxGenericAppDelegate* ofxGenericApp::getAppDelegate()
 
 void ofxGenericApp::runViaInfiniteLoop( ofPtr< ofxAppGenericWindow > window )
```
Comment (erynofwales) [inline src/ofxGenericApp.cpp]: This method doesn't actually loop at all...

File: src/openFrameworks/ofBaseApp.h [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+		virtual void mouseReleased(){}
+		virtual void mouseReleased(int x, int y, int button ){}
+		
+		// IOSP-40 // virtual void dragEvent(ofDragInfo dragInfo) { }
+		// IOSP-40 // virtual void gotMessage(ofMessage msg){ }
+	
+		virtual void windowEntry ( int state ) { }
+		
+		int mouseX, mouseY;			// for processing heads
+};
```
Comment (erynofwales) [inline src/openFrameworks/ofBaseApp.h]: What's going on with the formatting of this class? Is this a GitHub thing?

```
class ofBaseApp
{
    // Four space indents
    virtual ~ofBaseApp() {}
};
```

File: src/openFrameworks/ofColor.cpp [hunk 5]
```diff
@@ -0,0 +1,710 @@
...
+template<typename PixelType> const ofColor_<PixelType> ofColor_<PixelType>::thistle(0.847059*limit(),0.74902*limit(),0.847059*limit());
+template<typename PixelType> const ofColor_<PixelType> ofColor_<PixelType>::tomato(1*limit(),0.388235*limit(),0.278431*limit());
+template<typename PixelType> const ofColor_<PixelType> ofColor_<PixelType>::turquoise(0.25098*limit(),0.878431*limit(),0.815686*limit());
+template<typename PixelType> const ofColor_<PixelType> ofColor_<PixelType>::violet(0.933333*limit(),0.509804*limit(),0.933333*limit());
+template<typename PixelType> const ofColor_<PixelType> ofColor_<PixelType>::wheat(0.960784*limit(),0.870588*limit(),0.701961*limit());
+template<typename PixelType> const ofColor_<PixelType> ofColor_<PixelType>::whiteSmoke(0.960784*limit(),0.960784*limit(),0.960784*limit());
+template<typename PixelType> const ofColor_<PixelType> ofColor_<PixelType>::yellowGreen(0.603922*limit(),0.803922*limit(),0.196078*limit());
+template<typename PixelType> const ofColor_<PixelType> ofColor_<PixelType>::clearColor( limit(), limit(), limit(), 0);
```
Comment (erynofwales) [inline src/openFrameworks/ofColor.cpp]: Wowwwwww... O_O

File: src/iOS/ofxGenericAppDelegate.h (hunk 1 above)
Comment (dterzic) [inline src/iOS/ofxGenericAppDelegate.h]: Reason I've commented-out (instead of removing) is that I wanted to be code reviewed  because some of that stuff might be actually needed (removed by mistake).

File: src/openFrameworks/ofColor.cpp (hunk 5 above)
Comment (dterzic) [inline src/openFrameworks/ofColor.cpp]: :) that file is ported (copied) directly from the openFrameworks

File: src/openFrameworks/ofBaseApp.h (hunk 4 above)
Comment (dterzic) [inline src/openFrameworks/ofBaseApp.h]: That class is originally part of the openFrameworks; I've just copied it directly.

File: src/iOS/ofxGenericAppDelegate.h [hunk 6]
```diff
@@ -8,10 +8,11 @@
 
 #pragma once
 
-#include "ofMain.h"
+// IOSP-40 // #include "ofMain.h"
```
Comment (patbarry) [inline src/iOS/ofxGenericAppDelegate.h]: seems like you can remove this.

File: src/iOS/ofxGenericAppDelegate.h (hunk 1 above)
Comment (patbarry) [inline src/iOS/ofxGenericAppDelegate.h]: why remove this?

File: src/iOS/ofxGenericAppDelegate.mm [hunk 7]
```diff
@@ -44,8 +37,9 @@ -( BOOL )application:( UIApplication* )application didFinishLaunchingWithOptions
     ofPtr< ofxGenericApp > app = ofxGenericApp::getInstance();
     
     // TODO: don't use ofSetupOpenGL
-    ofSetupOpenGL( window, 1024, 768, OF_FULLSCREEN );
-    ofRunApp( app );
+    // IOSP-40 // ofSetupOpenGL( window, 1024, 768, OF_FULLSCREEN );
+    // IOSP-40 // ofRunApp( app );
```
Comment (patbarry) [inline src/iOS/ofxGenericAppDelegate.mm]: looks good .. can remove these comments

File: src/ofxAppGenericWindow.h [hunk 8]
```diff
@@ -11,16 +11,17 @@
...
+// IOSP-40 // #include "ofAppBaseWindow.h"
 #endif
 
 #include "ofxGenericView.h"
 
 class ofxAppGenericWindow
 #if TARGET_OS_IPHONE
-: public ofAppiPhoneWindow
+// IOSP-40 //: public ofAppiPhoneWindow
+: public ofxGenericView
```
Comment (patbarry) [inline src/ofxAppGenericWindow.h]: shouldn't this inherit from ofxAppGenericWindow instead?

File: src/ofxGenericApp.cpp [hunk 9]
```diff
@@ -140,7 +143,10 @@ void ofxGenericApp::finishedLaunching()
     handleFinishedLaunchingPresetup();
     
     // wait a cycle so iOS has time to get initialized
-    _setupTimer = ofxGenericTimer::create( 0.0001f, false, dynamic_pointer_cast< ofxGenericTimerDelegate >( _this ) );
+    // IOSP-40 // _setupTimer = ofxGenericTimer::create( 0.0001f, false, dynamic_pointer_cast< ofxGenericTimerDelegate >( _this ) );
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, 1 * NSEC_PER_MSEC), dispatch_get_main_queue(), ^{
```
Comment (patbarry) [inline src/ofxGenericApp.cpp]: this delay now is faster than before

File: src/ofxGenericApp.cpp [hunk 10]
```diff
@@ -177,8 +183,8 @@ void ofxGenericApp::timer_fired( ofPtr< ofxGenericTimer > timer )
 {
     if ( timer == _setupTimer )
     {
-        ofNotifySetup();
-        ofNotifyUpdate();
+        // IOSP-40 // ofNotifySetup();
+        // IOSP-40 // ofNotifyUpdate();
```
Comment (patbarry) [inline src/ofxGenericApp.cpp]: you could delete this whole method

File: src/ofxGenericApp.cpp [hunk 11]
```diff
@@ -595,8 +601,8 @@ void ofNotifyDeviceOrientationChanged( ofOrientation orientation )
 //	}
 	
 #ifdef OF_USING_POCO
-    orientationEventArgs.orientation = orientation;
-    ofNotifyEvent( ofxGenericEvents.orientation, orientationEventArgs );
+    // IOSP-40 // orientationEventArgs.orientation = orientation;
+    // IOSP-40 // ofNotifyEvent( ofxGenericEvents.orientation, orientationEventArgs );
 #endif
```
Comment (patbarry) [inline src/ofxGenericApp.cpp]: just delete the ifdef then.

File: src/ofxGenericMain.h [hunk 12]
```diff
@@ -8,13 +8,15 @@
 
 #pragma once
 
-#include "ofMain.h"
+// IOSP-40 // #include "ofMain.h"
 
 #if TARGET_OS_IPHONE
-#import <UIKit/UIKit.h>
+
+// IOSP-40 // #import <UIKit/UIKit.h>
```
Comment (patbarry) [inline src/ofxGenericMain.h]: why remove this?

File: src/ofxGenericValueStore.mm [hunk 13]
```diff
@@ -18,6 +18,8 @@
 #include "ofxJSONElement.h"
 #include <tinyxml.h>
 
+#include "ofUtils.h"
+
```
Comment (patbarry) [inline src/ofxGenericValueStore.mm]: why did you change the name of this file?  it was already Objective-C++

**Comment by patbarry** [comment]
few small comments .. otherwise looks fine. I'm not too concerned whether you delete all those IOSP comments or not .. we're ditching this whole thing asap.
